### PR TITLE
Use external upload script

### PR DESCRIPTION
**Pull request by jakirkham**
Switches to using `upload_or_check_non_existence.py` from `conda-forge-build-setup`. This should help allow that script to evolve more quickly as appears to be needed. In the long term, we plan to eliminate it completely if possible.

This should also help alleviate the upload issues we have been experiencing as we will be able to leverage PR ( https://github.com/conda-forge/conda-forge-build-setup-feedstock/pull/26 ).

Edit: Testing re-rendering in PR ( https://github.com/conda-forge/splauncher-feedstock/pull/2 ).

### CODE REVIEW

File: conda_smithy/templates/appveyor.yml.tmpl [hunk 1]
```diff
@@ -64,6 +64,6 @@ build: off
 test_script:
     - "%CMD_IN_ENV% conda build {{ recipe_dir }} --quiet"
 deploy_script:
-{% for owner, channel in channels['targets'] %}
-    - 'python ci_support\upload_or_check_non_existence.py .\{{ recipe_dir }} {{ owner }} --channel={{ channel }}'
+{%- for owner, channel in channels['targets'] %}
+    - cmd: upload_or_check_non_existence.py .\{{ recipe_dir }} {{ owner }} --channel={{ channel }}
```
Comment (jakirkham) [inline conda_smithy/templates/appveyor.yml.tmpl]: Have tried a few variations of this and Windows likes none of them. Maybe we can add a batch script that calls the Python file IDK. Going to let someone else resolve that and we can try this again.

**Comment by jakirkham** [comment]
This works fine on Unix, but Windows needs some help.

**Comment by jakirkham** [comment]
After some more tweaking, this seems to work fine on AppVeyor. Think we are ready to go ahead with this.

**Comment by jjhelmus** [comment]
Is `python` needed before the upload script in the appveyor.yaml file?

**Comment by jjhelmus** [comment]
Nope it does not seem to be...

**Comment by jakirkham** [comment]
Nope. See this [build]( https://ci.appveyor.com/project/conda-forge/splauncher-feedstock/build/1.0.9/job/9wbdl0x5qxtbhqlg#L357 ) for an example.

**Comment by jjhelmus** [comment]
LGTM, but I'm can't merge...

**Comment by jjhelmus** [comment]
closes #268?

**Comment by jakirkham** [comment]
Not quite. That issue is more about total removal of this script from `conda-forge`.

For now we are settling for moving it out and whittling it down. Still we could open a new issue for that at `conda-forge-build-setup` and close that one out.

**Comment by jakirkham** [comment]
> LGTM, but I can't merge...

Thanks. Will try to give @pelson a chance to look at it.

How soon are you needing a `conda-smithy` release/what is the issue that is driving it?

**Comment by jjhelmus** [comment]
conda-forge/python-feedstock#70.  Every package we have created in the last week is nearly a meg larger than needed and I need conda-build 2.0.1+ to fix it.

**Comment by jjhelmus** [comment]
Alternatively if someone can move the  the python `3.5.2_2` build off the main label on conda-forge's channel that would also fix the issue for the time being.

**Comment by jakirkham** [comment]
Moving them off `main` is easy.

Is it build number `2` or `3` though? Also does this impact other versions or just `3.5.2`?

**Comment by jjhelmus** [comment]
Just 3.5.2 build 3, mistyped the last comment.  build 2 seems to be fine as it was build with a different version of conda-build.

**Comment by jakirkham** [comment]
No worries. Just to confirm this is all OSes?

**Comment by jakirkham** [comment]
Also there appear to be some build number 2 packages that were built in the last 5 days (just like the build number 3 packages). Should these be pulled as well?

**Comment by jakirkham** [comment]
I have labeled the build number 3 packages as `broken`. If some of them need to go back into, `main` we can move them if needed.

screen shot 2016-09-13 at 22 45 27

**Comment by jakirkham** [comment]
Do we need to pull any other ones, @jjhelmus, or was that it?

**Comment by jjhelmus** [comment]
That should be the only one that need to be pulled.  If I see anything else wrong I'll reach out.

**Comment by jakirkham** [comment]
Sorry for the tangential discussion, @pelson. If you would like me to put it in a fresh PR, that can be done.

Should mention this is the PR to move to using `conda-forge-build-setup` for the upload script. Would be nice to see this merged soon-ish, but I know we are all busy people. Just want to make sure it doesn't get lost.

**Comment by jakirkham** [comment]
Friendly nudge, @pelson.

Sorry to do it so soon after the last one. Wanted to do it while you were online in the hopes it doesn't get buried.

**Comment by pelson** [comment]
Please could you open another PR that removes the file from the feedstock if it already exists? I implemented such a thing in https://github.com/conda-forge/conda-smithy/pull/193/files#diff-3a2c4a048580db27ad9db93e2780b511R316 - we want that kind of code + the removal of the upload script.

Additionally, now is the time to rename this upload script. Something like `add_to_channel` perhaps?

**Comment by jakirkham** [comment]
Please see PR ( https://github.com/conda-forge/conda-smithy/pull/311 ) for handling removal of `upload_or_check_non_existence.py` from feedstocks.

**Comment by jakirkham** [comment]
> Additionally, now is the time to rename this upload script. Something like `add_to_channel` perhaps?

Not against renaming this script, but am not going to take this on personally.

**Pull request merged**